uplink/decorators.py

```diff
@@ -103,10 +103,13 @@ class headers(MethodAnnotation):
     """
 
     def __init__(self, arg, **kwargs):
-        # TODO: allow the first argument to be a list or str, in
-        # which case you would split the strings by a colon delimiter
-        # to identify the headers.
-        self._headers = dict(arg, **kwargs)
+        if isinstance(arg, str):
+            header = arg.replace(":", "").split(" ")
+            self._headers = {header[0]: header[1]}
+        elif isinstance(arg, list):
+            self._headers = {arg[0].replace(":", ""): arg[1]}
```

Comment:
A list should be in the format: `["key1: value1", "key2: value2", ...]`. This would create the headers dict:
```python
{"key1": "value1", "key2": "value2", ...}
``` 

In other words, when the argument is a list, each element should represent a key, value pair.